sql/parser/builtins.go

```diff
@@ -248,6 +248,33 @@ var builtins = map[string][]builtin{
 		return DString(strings.Replace(s, from, to, -1)), nil
 	}, DummyString)},
 
+	"translate": {stringBuiltin3(func(s, from, to string) (Datum, error) {
+		fromRunes := []rune(from)
```

Comment:
This can be done without creating quite so many copies. POC: http://play.golang.org/p/QionwkgK3k